### PR TITLE
feat: assert not exists topic

### PR DESCRIPTION
**Pull request by jzaralim**
### Description 
Adds `NOT EXISTS` to `ASSERT TOPIC` which asserts that a topic doesn't exist

### Testing done 
Manual, unit+integration tests

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/AssertTopicExecutor.java [hunk 1]
```diff
@@ -59,52 +62,75 @@ public static StatementExecutorResponse execute(
...
-        }
-      });
-      if (configErrors.size() > 0) {
-        throw new KsqlException(String.join("\n", configErrors));
+    if (!assertExists) {
+      if (config.size() > 0) {
+        LOG.warn("Will skip topic config check for topic non-existence assertion.");
+      }
+      if (topicExists) {
+        throw new KsqlException("Topic " + topic + " exists");
```
Comment (jnh5y) [inline ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/AssertTopicExecutor.java:98]: I'm 95% sure, but just to make sense, but this logic is instead the call which is retried, so if the topic is being deleted, the assertion can pass, etc, etc, right?  (Still waiting for my coffee to kick in.)

**Review by jnh5y — APPROVED** [review]
LGTM.

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/AssertTopicExecutor.java (hunk 1 above)
Comment (jzaralim) [reply ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/AssertTopicExecutor.java:98]: Yup, if this fails then the entire `assertTopic` method is retried until the timeout

**Comment by jzaralim** [comment]
> LGTM. (Once this is in, looks like we'd just need a doc PR.)

We also need to update the Java client and the migration tool, and we need to do all this again for `ASSERT SCHEMA` :)

**Pull request merged**